### PR TITLE
Possibility to use a commit id instead of branch name for file-linking

### PR DESCRIPTION
**Pull request by janniksam**
**What I have done:**

- I created two new classes Branch and Commit, that hold the relevant information now.
- I added new overloads to the `FileLinkSettings` and `Aliases.FileLinking` to allow to pass a commit instead of a branch when linking files in an Azure DevOps enviroment. The existing methods now get a branch-object instead of a string.

### CODE REVIEW

**Comment by janniksam** [comment]
I knew that you would have doubts about it. The reason why I did it is simple: To let both methods have the same name. And I didn't want to just add it as another argument to the existing one, because it would confuse the consumer on what is mandatory and what not (+ what would happen if both are filled). Thats why I decided to bring in some descriptive wrapper class around both strings and, for now, it would be an ok'ish solution.

__tl;dr:__

I can do that for sure. But right now, I do not know, if a fluent builder would be a bit "too much" for this rather simple setting-class.

**Comment by janniksam** [comment]
@pascalberger I just pushed the initial "builder". Can you please verify if this is the way you want to go (before I document everything and write a dozen tests)?

**Comment by janniksam** [comment]
I implemented it like it was mentioned above.

@pascalberger can you take a look again? And you should fix the grammar / spelling mistakes, I hate my english. ;-)

**Comment by janniksam** [comment]
Cleanup LGTM 👍

File: src/Cake.Issues/FileLinkSettings.cs [hunk 1]
```diff
@@ -81,5 +103,38 @@ public FileLinkSettings(string fileLinkPattern)
...
+
+            return
+                new FileLinkSettings(
+                    repositoryUrl.ToString().TrimEnd('/') +
+                    "?path=" + rootPath + "{FilePath}" +
+                    "&version=GC" + commitId +
+                    "&line={Line}" +
+                    "&lineEnd={EndLine}" +
+                    "&lineStartColumn={Column}" +
+                    "&lineEndColumn={EndColumn}");
```
Comment (pascalberger) [inline src/Cake.Issues/FileLinkSettings.cs:137]: If `EndLine`, `Column` or `EndColumn` is not available, link won't highlight the line. This will taken care of later in a rewrite of the `FileLinkSettings` class

**Comment by pascalberger** [comment]
@janniksam your changes have been merged, thanks for your contribution 👍
